### PR TITLE
chore: return minimal job data in sample responses

### PR DESCRIPTION
**Pull request by ryanfang5**

### CODE REVIEW

**Comment by ryanfang5** [comment]
Returned AnalysisSearchResult for GET in ("/samples/{sample_id}/analyses") - not sure if AnalysisMinimal should be returned for POST. For now I have only attached the JobMinimal data to the document.

File: tests/samples/test_api.py [hunk 1]
```diff
@@ -834,14 +834,26 @@ async def test_find_analyses(
...
+
+    await client.db.references.insert_many(
+        [
+            {"_id": "foo", "data_type": "genome", "name": "Foo"},
+            {"_id": "baz", "data_type": "genome", "name": "Baz"},
+        ],
+        session=None,
+    )
+
     await client.db.analyses.insert_many(
```
Comment (igboyes) [inline tests/samples/test_api.py:849]: Need a test for when the `job.id` field is set in the analysis, but no corresponding job is found in the database. This should result in the Pydantic field being set `None`.

File: virtool/samples/data.py [hunk 2]
```diff
@@ -159,6 +160,210 @@ async def find(
...
+                            "job.id": True,
+                            "job.archived": True,
+                            "ready": True,
+                            "reference": True,
+                            "sample": True,
+                            "subtractions": True,
+                            "updated_at": True,
+                            "user._id": True,
+                            "user.handle": True,
+                            "user.administrator": True,
```
Comment (igboyes) [inline virtool/samples/data.py:338]: Is there no way to simplify this. Can you not just do `"job": True` to cover all job fields?

**Review by igboyes — CHANGES_REQUESTED** [review]

File: virtool/samples/data.py (hunk 2 above)
Comment (ryanfang5) [reply virtool/samples/data.py]: I originally wanted to remove the "key" field from being returned in jobs, but I realised it was easier to do with an $unset. I also did this in analyses/data.

**Review by igboyes — APPROVED** [review]

**Pull request merged**